### ISSUE TITLE
4.0.0 Release discussion

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#1978]
## RELEASE VERSION

v4.0.0

### SUMMARY

This release will be primarily focused on adding dispensing functionality to mobile. This will be quite a quick turn around. Ideally we're looking at having everything merged into develop by the 27th of January!! With two weeks of testing until February 10th.

If anyone has any other features/bugs/refactors they think need to be in this release, or think we should discuss feel free to comment!

EDIT: Looking to be released between 20th-22nd February, after the desktop 4.09 release.

### Features

- [x] Dispensing [This feature contained 100's of issues, with internal tracking issues]
- [x] Supplier credits https://github.com/openmsupply/mobile/issues/1770 
- [x] Cash register https://github.com/openmsupply/mobile/issues/1917
- [x] Logout timer https://github.com/openmsupply/mobile/issues/2069 
- [x] Sell and price columns for incoming stock https://github.com/openmsupply/mobile/issues/1984 
- [x] Supplier column when editing a stock take batch https://github.com/openmsupply/mobile/issues/2142
- [x] Requisition reasons - https://github.com/openmsupply/mobile/issues/2205

### Refactors

N/A

### Bugs

N/A

### DOCUMENTATION

- [x] Dispensing 
    - DRAFT: https://docs.google.com/document/d/1ldvusBVEiTUXBrhUpxFh5Wff0s04AX59q7Ah-j1rO9g/edit
    - FINAL: https://docs.msupply.foundation/en:mobile:user_guide:dispensing
- [x] Supplier credits
    - DRAFT: https://docs.google.com/document/d/1IQuGgQXZqma8lLfYQtH-cqxM7w2rttE48YisuJ4JEEw/edit
    - FINAL: https://docs.msupply.foundation/en:mobile:user_guide:supplier_invoices_returns
- [x] Cash register 
    - DRAFT: https://docs.google.com/document/d/1r4STFAa3aoMwDl0InZbHUphgrPFQPSvOXmRNX88OCzo/edit
    - FINAL: https://docs.msupply.foundation/en:mobile:user_guide:cash_register
- [x] Logout timer
    - DRAFT: https://docs.google.com/document/d/1V95Qr4gm4ucJjzB-h7GoReu6TUE6qflpfELAz1AcSus/edit
    - FINAL: https://docs.msupply.foundation/en:mobile:user_guide:getting_started
- [x] Requisition reasons
    - DRAFT:  https://docs.google.com/document/d/1mSBbgl1NpfcCWjnJZRYQPThixU8qJQEexHJaf_EafBw/edit#
    - FINAL: https://docs.msupply.foundation/en:mobile:user_guide:supplier_requisitions_program
- [x] Setting up dispensary
    - DRAFT: https://docs.google.com/document/d/1B2f7_6eeISYL-L55hq-cOc0LmBVnpnIhp9FKHqRbwJs/edit
    - FINAL: https://docs.msupply.foundation/en:mobile:setup:server_side:config

**Comment by joshxg** [comment]
UPDATE: 

Have released an RC version for testing. Have also updated this version to 4.0.0 from 3.3.0 as there are fields which are not backwards compatible!

**Comment by joshxg** [comment]
UPDATE:

- Most testing has been completed on `#feature-dispensary`
- Looking to merge this into `develop` within the next couple of days.
- As 4.0.0 is backwards incompatible with any of the currently released `mSupply` versions, there doesn't seem to be much point in releasing it until after desktop 4.09 is released which is looking to be this week. Once that is done we should be able to release 4.0.0.

**Comment by joshxg** [comment]
UPDATE:

- Due to Desktop `4.09` being pushed back to 1st March, date of this release also required to be pushed back.

**Comment by joshxg** [comment]
UPDATE:

- The development has been consolidated to the develop branch.
- 4.09 is due to be released Friday 28th Feb NZT
- Due to there being changes occurring in the desktop development, there are still issues which need testing to ensure they are mobile compatible.
- If the testing brings up some issues which we need to fix on the mobile side, the absolute latest `4.0.0` will be released is Monday 2nd March

**Comment by joshxg** [comment]
UPDATE:

4.0.0 Has been released 🎉 

Somewhat prematurely as mSupply 4.09 has not been released just yet. Oops!

We'll keep this issue open as there is still a bit of documentation to complete.

Please check out the patch discussion issue here: https://github.com/openmsupply/mobile/issues/2444 where we can discuss immediate issues, if any, with the release.

Also #2445 for the next minor release! Feel free to comment and give suggestions!

**Comment by joshxg** [comment]
Docs all done 🎉 

Closing this one now - thanks everyone!
